tests/test_dfs_edges.py

```diff
@@ -0,0 +1,45 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import retworkx
+
+
+class TestDfsEdges(unittest.TestCase):
+    def test_graph_dfs_edges(self):
+        graph = retworkx.PyGraph()
+        graph.extend_from_edge_list([(0, 1), (1, 2), (1, 3), (2, 4), (3, 4)])
+        edges = retworkx.graph_dfs_edges(graph, 0)
+        expected = [(0, 1), (1, 2), (2, 4), (4, 3)]
+        self.assertEqual(expected, edges)
+
+    def test_digraph_dfs_edges(self):
+        graph = retworkx.PyDiGraph()
+        graph.extend_from_edge_list([(0, 1), (1, 2), (1, 3), (2, 4), (3, 4)])
+        edges = retworkx.digraph_dfs_edges(graph, 0)
+        expected = [(0, 1), (1, 2), (2, 4), (4, 3)]
```

Comment:
I'm confused. Is `4 -> 3` in this graph? I'd have expected `[(0, 1), (1, 2), (2, 4), (1, 3)]`

Reply:
Yeah, you're right this looks like a bug in the function. There shouldn't be an edge from 4->3, it should be 1->3. I guess I just blindly copy and pasted the results here when I originally wrote this and didn't think about it.

Reply:
Ok, I just pushed a fix for this here: https://github.com/Qiskit/retworkx/pull/146/commits/ae22f229acbd1356109e13fe5eb98328b7f1ef01.

The previous algorithm only worked if there was a linear series of edges, but would fall apart when the dfs order had nodes at a different level next to each other, as this showed. Basically it was incorrectly assuming that the previous entry in the dfs node list was always the parent for the edge, which wasn't always true. That commit manually does the dfs traversal and tracks the parents as it does so we can build the correct edge list.